scripts/gen-ceph-kustomize.sh

```diff
@@ -0,0 +1,213 @@
+     name: ceph
+     env:
+     - name: MON_IP
+       value: "$MON_IP"
+     - name: NETWORK_AUTO_DETECT
+       value: "$NET_DISCOVERY"
+     - name: CEPH_DAEMON
+       value: demo
+     - name: CEPH_PUBLIC_NETWORK
+       value: "0.0.0.0/0"
+     - name: DEMO_DAEMONS
+       value: "osd"
+     volumeMounts:
+      - mountPath: /var/lib/ceph
+        name: data
+      - mountPath: /var/log/ceph
+        name: log
+      - mountPath: /run/ceph
+        name: run
+  volumes:
+  - name: data
+    emptyDir:
+      sizeLimit: 500Mi
+  - name: run
+    emptyDir:
+      sizeLimit: 500Mi
+  - name: log
+    emptyDir:
+      sizeLimit: 500Mi
+  securityContext:
+    runAsUser: 0
+    seccompProfile:
+      type: Unconfined
+EOF
+}
+
+function ceph_kustomize {
+cat <<EOF >kustomization.yaml
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+resources:
+- ./ceph-pod.yaml
+namespace: openstack
+patches:
+- target:
+    kind: Pod
+  patch: |-
+    - op: replace
+      path: /metadata/namespace
+      value: $NAMESPACE
+    - op: replace
+      path: /spec/containers/0/image
+      value: $IMAGE
+    - op: replace
+      path: /metadata/annotations/k8s.v1.cni.cncf.io~1networks
+      value: $NETWORKS_ANNOTATION
+EOF
+}
+
+function get_pod_ip {
+    if [[ "$HOSTNETWORK" == "false" ]]; then
+        echo "HOSTNETWORK must be set to true!"
+        exit 1
+    fi
+    MON_IP=$(oc get nodes -o 'jsonpath={.items[*].status.addresses[?(.type=="InternalIP")].address}')
+    NET_DISCOVERY="0"
+}
+
+function ceph_is_ready {
+    echo "Waiting the cluster to be up"
+    until oc rsh ceph ls /etc/ceph/I_AM_A_DEMO &> /dev/null
+    do
+        sleep 1
+        echo -n .
+        (( TIMEOUT-- ))
+        [[ "$TIMEOUT" -eq 0 ]] && exit 1
+    done
+    echo
+}
+
+function create_pool {
+
+    [ "${#POOLS[@]}" -eq 0 ] && return;
+
+    for pool in "${POOLS[@]}"; do
+        oc rsh ceph ceph osd pool create $pool 4
+        oc rsh ceph ceph osd pool application enable $pool rbd
+    done
+}
+
+function build_caps {
+    local CAPS=""
+    for pool in "${POOLS[@]}"; do
+        caps="allow rwx pool="$pool
+        CAPS+=$caps,
+    done
+    echo "${CAPS::-1}"
+}
+
+function create_key {
+    local client=$1
+    local caps
+    local osd_caps
+
+    if [ "${#POOLS[@]}" -eq 0 ]; then
+        osd_caps="allow *"
+    else
+        caps=$(build_caps)
+        osd_caps="allow class-read object_prefix rbd_children, $caps"
+    fi
+    # do not log the key if exists
+    oc rsh ceph ceph auth get-or-create "$client" mgr "allow rw" mon "allow r" osd "$osd_caps" >/dev/null
+}
+
+function create_secret {
+
+    SECRET_NAME="$1"
+
+    TEMPDIR=`mktemp -d`
+    local client="client.openstack"
+    trap 'rm -rf -- "$TEMPDIR"' EXIT
+    echo 'Copying Ceph config files from the container to $TEMPDIR'
+    oc rsync ceph:/etc/ceph/ceph.conf $TEMPDIR
+    echo 'Create OpenStack keyring'
+    # we build the cephx openstack key
+    create_key "$client"
```

Comment:
@ASBishop @fultonj instead of using the admin keyring, we're building a new `client.openstack` key and updating the secret accordingly.
We can expose more parameters, but in order to keep things simple I'd just use the `openstack` user. In case we need more we can follow up.